Lib/fontTools/feaLib/builder.py

```diff
@@ -494,6 +500,176 @@ def checkattr(table, attrs):
         if version >= 5:
             checkattr(table, ("usLowerOpticalPointSize", "usUpperOpticalPointSize"))
 
+    def set_ElidedFallbackName(self, names):
+        self.stat_["ElidedFallbackName"] = names
+
+    def set_ElidedFallbackNameID(self, value):
+        self.stat_["ElidedFallbackNameID"] = value
+
+    def addDesignAxis(self, location, key, axisOrder, value):
+        if "DesignAxes" not in self.stat_:
+            self.stat_["DesignAxes"] = {}
+        if key in self.stat_["DesignAxes"]:
+            raise FeatureLibError(
+                'DesignAxis already defined for tag "%s".' % key,
+                location,
+            )
+        self.stat_["DesignAxes"][key] = (axisOrder, value)
+
+    def addAxisValueRecord(self, location, names, locations, flags):
+        if "AxisValueRecords" not in self.stat_:
+            self.stat_["AxisValueRecords"] = []
+        # Check for duplicate AxisValueRecords
+        for record in self.stat_["AxisValueRecords"]:
+            names_, locations_, flags_ = record
+            if sorted(names_) == sorted(names) and sorted(locations_) == \
+                    sorted(locations) and sorted(flags_) == sorted(flags):
+                raise FeatureLibError(
+                    'Duplicate AxisValueRecords are not allowed.',
+                    location,
+                )
+        self.stat_["AxisValueRecords"].append((names, locations, flags))
+
+    def build_STAT(self):
```

Comment:
We recently got `otlLib.builder.buildStatTable()`, can’t it be used here?

Reply:
It could be used but the logic of `otLib.build.buildStatTable()` is rather different from what FEA syntax does, which is closer to what AxisValueRecords do, and it also doesn’t support languageIDs.
Basically `otLib.build.buildStatTable()` was written to be easy to use, but is not easy to use with FEA syntax.

Reply:
Thanks Khaled for reminding me of that. Yes, we should definitely try to reuse that code in here too, see whether it's possible to modify it to accommodate both.

Reply:
@punchcutter have you tried to see if there's anything in otlLib that you can reuse in here?
https://github.com/fonttools/fonttools/blob/bec19252c5b424a88e1dfeb423fb2c5068fb36e9/Lib/fontTools/otlLib/builder.py#L2399-L2590

Reply:
The base build code was taken from there since it was already there. Of course it's possible to merge the two, but I wanted to get this working in feaLib first before messing with that.

Reply:
yes I agree

Reply:
I also agree, and would be happy to review changes or refactorings to `fontTools.otlLib.builder.buildStatTable()` to make this possible.

Reply:
Ok, my initial thought was to get this in ASAP so we'd have it for some existing projects that are already set up with STAT in the feature file. And then after that definitely do a refactoring at some point. Since this is already quite a lot of new stuff I felt like it might be good to leave the otlLib code alone for a first round, but either way is fine.